releases/release-5.1.1.md

```diff
@@ -0,0 +1,113 @@
+
++ TiKV
+
+    - Make prewrite as idempotent as possible to reduce the chance of undetermined errors. [#10586](https://github.com/tikv/tikv/pull/10586)
+    - Fix duration calculation panics on certain platforms [#10569](https://github.com/tikv/tikv/pull/10569)
+    - Fix unencoded keys of `batch_get_command` in load-base-split [#10565](https://github.com/tikv/tikv/pull/10565)
+    - Prevent the risk of stack overflow when handling many expired commands. [#10502](https://github.com/tikv/tikv/pull/10502)
+    - Fix online changing `resolved-ts.advance-ts-interval` can't take effect immediately [#10494](https://github.com/tikv/tikv/pull/10494)
+    - fix follower meta corruption in rare cases with more than 4 replicas [#10486](https://github.com/tikv/tikv/pull/10486)
+    - Avoid panic when building a snapshot twice if encryption enabled [#10464](https://github.com/tikv/tikv/pull/10464)
+    - Not use the stale read request's `start_ts` to update `max_ts` to avoid commit request keep retrying [#10451](https://github.com/tikv/tikv/pull/10451)
+    - Fix wrong tikv_raftstore_hibernated_peer_state metric. [#10432](https://github.com/tikv/tikv/pull/10432)
+    - copr: fix the wrong arguments type of json_unquote [#10428](https://github.com/tikv/tikv/pull/10428)
+
++ PD
+
+    - TiDB Dashboard: Add OIDC based SSO support [#3883](https://github.com/tikv/pd/pull/3883)
+    - Reduce the conflict due to multiple scheduler running in same time [#3857](https://github.com/tikv/pd/pull/3857)
+    - Fix the issue that the scheduler may appear again even if we have already executed the delete operation [#3824](https://github.com/tikv/pd/pull/3824)
+
++ Tools
+
+    + TiCDC
+
+        - Remove file sorter. [#2327](https://github.com/pingcap/ticdc/pull/2327)
+        - puller,mounter,processor: always pull the old value internally [#2306](https://github.com/pingcap/ticdc/pull/2306)
+        - Fix minor runtime panic risk [#2300](https://github.com/pingcap/ticdc/pull/2300)
+        - Fix potential DDL loss when owner crashes while executing DDL [#2290](https://github.com/pingcap/ticdc/pull/2290)
+        - Don't resolve lock immediately after a region is initialized. [#2266](https://github.com/pingcap/ticdc/pull/2266)
+        - Fix potential DDL loss when owner crashes while executing DDL [#2252](https://github.com/pingcap/ticdc/pull/2252)
+        - Don't resolve lock immediately after a region is initialized. [#2235](https://github.com/pingcap/ticdc/pull/2235)
+        - Better err msg when PD endpoint missing certificate [#2186](https://github.com/pingcap/ticdc/pull/2186)
+        - Fix a bug that could cause data losses if a TiCDC node is kills immediate after a table migration [#2033](https://github.com/pingcap/ticdc/pull/2033)
+        - Fix changefeed update to properly handle --sort-dir and --start-ts. [#1921](https://github.com/pingcap/ticdc/pull/1921)
```

Comment:
TiCDC:

Improvements
- Remove file sorter. [#2327](https://github.com/pingcap/ticdc/pull/2327)
- Better err msg when PD endpoint missing certificate [#2186](https://github.com/pingcap/ticdc/pull/2186)

Bug fixes
- Fix support for new collation [#2306](https://github.com/pingcap/ticdc/pull/2306)
- Fix minor runtime panic risk [#2300](https://github.com/pingcap/ticdc/pull/2300)
- Fix potential DDL loss when owner crashes while executing DDL [#2290](https://github.com/pingcap/ticdc/pull/2290)
- Fix trying to resolve locks prematurely [#2266](https://github.com/pingcap/ticdc/pull/2266)
- Fix a bug that could cause data losses if a TiCDC node is killed immediately after a table migration [#2033](https://github.com/pingcap/ticdc/pull/2033)
- Fix changefeed update to properly handle --sort-dir and --start-ts. [#1921](https://github.com/pingcap/ticdc/pull/1921)

Reply:
BR:

Bug fixes
- Fix incorrectly calculating the size of data to restore [#1285](https://github.com/pingcap/br/pull/1285)
- Fix missed DDL events when restoring from cdclog [#1094](https://github.com/pingcap/br/pull/1094

Reply:
Lightning:

Improvements:
- Add retry for restoring schemas [#1294](https://github.com/pingcap/br/pull/1294)

Bug fixes
- Fix parquet parser for decimal type [#1272](https://github.com/pingcap/br/pull/1272)
- Fix integer overflows when calculating key intervals [#1294](https://github.com/pingcap/br/pull/1294)